codebuild/bin/grep_simple_mistakes.sh

```diff
@@ -77,6 +77,18 @@ for file in $S2N_FILES_ASSERT_NOTNULL_CHECK; do
   done < <(grep -rnE -A 1 "=\ss2n_stuffer_raw_read\(.*\)" $file)
 done
 
+# Assert that "index" is not a variable name. An "index" function exists in strings.h, and older compilers (<GCC 4.8) 
+# warn if any local variables called "index" are used because they are considered to shadow that declaration. 
+S2N_FILES_ASSERT_VARIABLE_NAME_INDEX=$(find "$PWD" -type f -name "s2n*.[ch]")
+for file in $S2N_FILES_ASSERT_VARIABLE_NAME_INDEX; do
+
+RESULT_VARIABLE_NAME_INDEX=`grep -v ' *[\/|\* ][\*|\/]' $file | grep ' \**index[;|,|\)| ]'`
+  if [ "${#RESULT_VARIABLE_NAME_INDEX}" != "0" ]; then
```

Comment:
Updating to: `gcc -fpreprocessed -dD -E $file | grep -v '"' | grep '[\*|,|;|[:space:]]index[;|,|\)|[:space:]]'`